roles/setup/tasks/main.yml

```diff
@@ -1,2 +1,108 @@
 ---
 # tasks file for setup
+
+- name: Check for CLUSTER_ALIAS input
+  fail:
+    msg: "You must supply '-e CLUSTER_ALIAS=<cluster_alias>'."
+  when:
+    - not CLUSTER_ALIAS is defined
+
+- name: Check for CLUSTER_API input
+  fail:
+    msg: "You must supply '-e CLUSTER_API=<cluster_api>'."
+  when:
+    - not CLUSTER_API is defined
+
+- name: Check for CLUSTER_USER input
+  fail:
+    msg: "You must supply '-e CLUSTER_USER=<cluster_user>'."
+  when:
+    - not CLUSTER_USER is defined
+
+- name: Check for CLUSTER_TOKEN_PATH input
+  fail:
+    msg: "You must supply '-e CLUSTER_TOKEN_PATH=</path/to/file/with/ocp/token>'."
+  when:
+    - not CLUSTER_TOKEN_PATH is defined
+
+- name: Check that CLUSTER_TOKEN_PATH exists
+  stat:
+    path: "{{ CLUSTER_TOKEN_PATH }}"
+  register: cluster_token_stat_result
+
+- name: CLUSTER_TOKEN_PATH does not exits
+  fail:
+    msg: "No file was found for CLUSTER_TOKEN_PATH at '{{ CLUSTER_TOKEN_PATH }}'."
+  when:
+    - not cluster_token_stat_result.stat.exists
+
+- name: Generate random UUID for cluster identifier
+  set_fact:
+    cluster_id: "{{ 9999999999999999999999 | random | to_uuid }}"
+
+- name: Define OCP Usage configuration defaults
+  set_fact:
+    ocp_usage_config_dir: "{{ ansible_env.HOME }}/.config/ocp_usage"
+    ocp_usage_config_file: "{{ ansible_env.HOME }}/.config/ocp_usage/config.json"
+  when:
+    - OCP_USAGE_CONFIG_DIR is not defined
+
+- name: Define OCP Usage configuration
+  set_fact:
+    ocp_usage_config_dir: "{{ OCP_USAGE_CONFIG_DIR }}"
+    ocp_usage_config_file: "{{ OCP_USAGE_CONFIG_DIR }}/config.json"
+  when:
+    - OCP_USAGE_CONFIG_DIR is defined
+
+- name: Create config data location
+  file:
+    path: "{{ ocp_usage_config_dir }}"
+    state: directory
+    mode: 0755
+
+- name: Check that OCP Usage configuration exists
+  stat:
+    path: "{{ ocp_usage_config_file }}"
+  register: ocp_usage_config_stat_result
+
+- name: Read OCP Usage configuration contents
+  command: cat {{ ocp_usage_config_file }}
```

Comment:
This isn't necessary. In the `copy` task, you can use `with_file` to read the file contents, like this:

```
- name: Transform OCP Usage configuration contents
  set_fact:
    config_content: "{{ item | from_json }}"
  when:
    - ocp_usage_config_stat_result.stat.exists
    - ocp_usage_config_file_cat.stdout is defined
    - OCP_USAGE_CONFIG_OVERWRITE is defined
    - "{{ OCP_USAGE_CONFIG_OVERWRITE | bool }}"
  with_file:
    - /your/file/here
```

Reply:
Hit a problem with lookup file mechanism so worked around it with cat:
```
    TASK [setup : Check that OCP Usage configuration exists] ***********************
    task path: /Users/christopherhambridge/Code/korekuta/roles/setup/tasks/main.yml:63
    <instance> ESTABLISH DOCKER CONNECTION FOR USER: root
    <instance> EXEC ['/usr/local/bin/docker', b'exec', b'-i', 'instance', '/bin/sh', '-c', "/bin/sh -c 'echo ~ && sleep 0'"]
    <instance> EXEC ['/usr/local/bin/docker', b'exec', b'-i', 'instance', '/bin/sh', '-c', '/bin/sh -c \'( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1539878543.0233011-259806788198958 `" && echo ansible-tmp-1539878543.0233011-259806788198958="` echo /root/.ansible/tmp/ansible-tmp-1539878543.0233011-259806788198958 `" ) && sleep 0\'']
    Using module file /Users/christopherhambridge/.local/share/virtualenvs/korekuta-1YD3gLAc/lib/python3.6/site-packages/ansible/modules/files/stat.py
    <instance> PUT /Users/christopherhambridge/.ansible/tmp/ansible-local-10569wozilu5r/tmp27x197lw TO /root/.ansible/tmp/ansible-tmp-1539878543.0233011-259806788198958/AnsiballZ_stat.py
    <instance> EXEC ['/usr/local/bin/docker', b'exec', b'-i', 'instance', '/bin/sh', '-c', "/bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1539878543.0233011-259806788198958/ /root/.ansible/tmp/ansible-tmp-1539878543.0233011-259806788198958/AnsiballZ_stat.py && sleep 0'"]
    <instance> EXEC ['/usr/local/bin/docker', b'exec', b'-i', 'instance', '/bin/sh', '-c', "/bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1539878543.0233011-259806788198958/AnsiballZ_stat.py && sleep 0'"]
    <instance> EXEC ['/usr/local/bin/docker', b'exec', b'-i', 'instance', '/bin/sh', '-c', "/bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1539878543.0233011-259806788198958/ > /dev/null 2>&1 && sleep 0'"]
    ok: [instance] => {
        "changed": false,
        "invocation": {
            "module_args": {
                "checksum_algorithm": "sha1",
                "follow": false,
                "get_attributes": true,
                "get_checksum": true,
                "get_md5": null,
                "get_mime": true,
                "path": "/etc/ocp_usage/config.json"
            }
        },
        "stat": {
            "atime": 1539878534.220036,
            "attr_flags": "",
            "attributes": [],
            "block_size": 4096,
            "blocks": 8,
            "charset": "unknown",
            "checksum": "135f4b86b1eb8ad08c43e8ff3e3e3b03202c2a2e",
            "ctime": 1539878534.220036,
            "dev": 2049,
            "device_type": 0,
            "executable": false,
            "exists": true,
            "gid": 0,
            "gr_name": "root",
            "inode": 1192757,
            "isblk": false,
            "ischr": false,
            "isdir": false,
            "isfifo": false,
            "isgid": false,
            "islnk": false,
            "isreg": true,
            "issock": false,
            "isuid": false,
            "mimetype": "unknown",
            "mode": "0644",
            "mtime": 1539878533.390036,
            "nlink": 1,
            "path": "/etc/ocp_usage/config.json",
            "pw_name": "root",
            "readable": true,
            "rgrp": true,
            "roth": true,
            "rusr": true,
            "size": 163,
            "uid": 0,
            "version": null,
            "wgrp": false,
            "woth": false,
            "writeable": true,
            "wusr": true,
            "xgrp": false,
            "xoth": false,
            "xusr": false
        }
    }
    
    TASK [setup : Read OCP Usage configuration contents] ***************************
    task path: /Users/christopherhambridge/Code/korekuta/roles/setup/tasks/main.yml:68
 [WARNING]: Unable to find '/etc/ocp_usage/config.json' in expected paths (use
-vvvvv to see paths)

    fatal: [instance]: FAILED! => {
        "msg": "An unhandled exception occurred while running the lookup plugin 'file'. Error was a <class 'ansible.errors.AnsibleError'>, original message: could not locate file in lookup: /etc/ocp_usage/config.json"
    }
```